src/main/java/com/hubspot/jinjava/util/LengthLimitingStringBuilder.java

```diff
@@ -50,14 +51,41 @@ public void append(Object obj) {
     append(String.valueOf(obj));
   }
 
-  public void append(String str) {
-    if (str == null) {
-      return;
+  @Override
+  public LengthLimitingStringBuilder append(CharSequence csq) {
+    int csqLength = 4; // null
+    if (csq != null) {
+      csqLength = csq.length();
+    }
+    length += csqLength;
+    checkLength();
+    builder.append(csq);
+    return this;
+  }
+
+  @Override
+  public Appendable append(CharSequence csq, int start, int end) {
+    int csqLength = 4; // null
+    if (csq != null) {
+      csqLength = end - start;
```

Comment:
Do you need to check if `end >= start`?

Reply:
`StringBuilder` will throw an IndexOutOfBoundsException if they're invalid so we don't need to